server/app/repository/ApplicationEventRepository.java

```diff
@@ -62,4 +67,27 @@ public ImmutableList<ApplicationEventModel> getEventsOrderByCreateTimeDesc(Long
                 queryProfileLocationBuilder.create("getEventsOrderByCreateTimeDesc"))
             .findList());
   }
+
+  public void setNote(
+      ApplicationModel application, ApplicationEventDetails.NoteEvent note, AccountModel admin) {
+    ApplicationEventDetails details =
+        services.application.ApplicationEventDetails.builder()
+            .setEventType(ApplicationEventDetails.Type.NOTE_CHANGE)
+            .setNoteEvent(note)
+            .build();
+    ApplicationEventModel event =
+        new ApplicationEventModel(application, Optional.of(admin), details);
+    try (Transaction transaction = database.beginTransaction(TxIsolation.SERIALIZABLE)) {
```

Comment:
**question**: Overall this is much less likely to have issues keeping the tables in sync, thank you!

There is one additional race condition that we may actually not care about, but it's this:
- What if someone submits a new application between when the `applicantModel` passed into this class was retrieved, and when this note is set? This transaction will now prevent the other issue of a note not getting copied to an application, but it doesn't prevent a note getting set on what is no longer the newest application.
- Is that actually a problem though? I think it was a possibility even with the old code, so it's probably not a problem, right?

Reply:
it doesn't prevent a note getting set on what is no longer the newest application. --> Imo, I don't think it is a problem. In the system today, we have notes and statuses associated with old applications. If a new application gets submitted, the programs admins will use the note feature to mark it as "duplicate" until we can find a solution to merge two applications and highlight the difference.